document-store/src/main/java/org/hypertrace/core/documentstore/postgres/query/v1/vistors/PostgresFilterTypeExpressionVisitor.java

```diff
@@ -81,16 +96,108 @@ public String visit(final KeyExpression expression) {
             postgresQueryParser.getParamsBuilder());
   }
 
-  @SuppressWarnings("unchecked")
+  @SuppressWarnings({"unchecked", "SwitchStatementWithTooFewBranches"})
   @Override
   public String visit(final ArrayRelationalFilterExpression expression) {
-    throw new UnsupportedOperationException();
+    /*
+    EXISTS
+     (SELECT 1
+      FROM jsonb_array_elements(COALESCE(planets->'elements', '[]'::jsonb)) AS elements
+      WHERE TRIM('"' FROM elements::text) = 'Oxygen'
+     )
+     */
+    switch (expression.getOperator()) {
+      case ANY:
+        // Convert 'elements' to planets->'elements' where planets could be an alias for an upper
+        // level array filter
+        // Also, for the first time (if this was not under any nesting), use the field identifier
+        // visitor to make it document->'elements'
```

Comment:
is this `document->'elements'` correct? should it be `document->'planets'?

Reply:
`document->'elements'` is correct. I've updated the code comments to be more clear.